src/zxing/zxing/qrcode/encoder/MaskUtil.cpp

```diff
@@ -53,30 +53,30 @@ int MaskUtil::applyMaskPenaltyRule3(const ByteMatrix& matrix)
 {
     int numPenalties = 0;
     const std::vector<std::vector<zxing::byte> >& array = matrix.getArray();
-    int width = matrix.getWidth();
-    int height = matrix.getHeight();
+    int width = int(matrix.getWidth());
+    int height = int(matrix.getHeight());
     for (int y = 0; y < height; y++) {
```

Comment:
No, we cannot change the type of x and y in the loop: in the calls to isWhiteHorizontal() and isWhiteVertical(), 4 is subtracted from x and y respectively. This would underflow if the variables were of unsigned type (they both start at zero).